packages/trie/prune.go

```diff
@@ -0,0 +1,47 @@
+package trie
+
+type PruneStats struct {
+	DeletedNodes  uint
+	DeletedValues uint
+}
+
+func Prune(store KVStore, trieRoot Hash) (PruneStats, error) {
+	refcounts := newRefcounts(store)
+
+	stats := PruneStats{}
+
+	tr, err := NewTrieReader(store, trieRoot)
+	if err != nil {
+		return stats, err
+	}
+
+	var deletedNodes []Hash
+	var deletedValues [][]byte
+
+	tr.IterateNodes(func(nodeKey []byte, n *NodeData, depth int) IterateNodesAction {
+		deleteNode, deleteValue := refcounts.Dec(n)
+		if deleteNode {
+			deletedNodes = append(deletedNodes, n.Commitment)
+		}
+		if deleteValue {
+			deletedValues = append(deletedValues, n.Terminal.Bytes())
+		}
+		if deleteNode {
+			return IterateContinue
+		}
```

Comment:
could be simplified like this?
```suggestion
		if deleteValue {
			deletedValues = append(deletedValues, n.Terminal.Bytes())
		}
		if deleteNode {
			deletedNodes = append(deletedNodes, n.Commitment)
			return IterateContinue
		}
```